src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -30,51 +28,51 @@
     public static final String COMMAND_WORD = "edit";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the module identified "
-            + "by the index number used in the displayed module list. "
-            + "Existing values will be overwritten by the input values.\n"
-            + "Parameters: INDEX (must be a positive integer) "
-            + "[" + PREFIX_CODE + "MODULE CODE] "
-            + "[" + PREFIX_CREDITS + "MODULAR CREDITS] "
-            + "[" + PREFIX_TAG + "TAG]...\n"
-            + "Example: " + COMMAND_WORD + " 1 "
-            + PREFIX_CREDITS + "4 ";
+        + "by its module code. "
+        + "Existing values will be overwritten by the input values.\n"
+        + "Parameters: MODULE CODE "
+        + "[" + PREFIX_CODE + "MODULE CODE] "
+        + "[" + PREFIX_TAG + "TAG]...\n"
+        + "Example: " + COMMAND_WORD + " cs2103t "
+        + PREFIX_CODE + "cs2108 ";
 
     public static final String MESSAGE_EDIT_MODULE_SUCCESS = "Edited Module: %1$s";
     public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
     public static final String MESSAGE_DUPLICATE_MODULE = "This module already exists in the GradPad.";
+    public static final String MESSAGE_MODULE_NOT_YET_ADDED = "Module %1$s has not been added yet!";
 
-    private final Index index;
+    private final ModuleCode code;
     private final EditModuleDescriptor editModuleDescriptor;
 
     /**
-     * @param index of the module in the filtered module list to edit
+     * @param code of the module in GradPad to edit
      * @param editModuleDescriptor details to edit the module with
      */
-    public EditCommand(Index index, EditModuleDescriptor editModuleDescriptor) {
-        requireNonNull(index);
+    public EditCommand(ModuleCode code, EditModuleDescriptor editModuleDescriptor) {
+        requireNonNull(code);
         requireNonNull(editModuleDescriptor);
 
-        this.index = index;
+        this.code = code;
         this.editModuleDescriptor = new EditModuleDescriptor(editModuleDescriptor);
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Module> lastShownList = model.getFilteredModuleList();
+        List<Module> modules = model.getGradPad().getModuleList();
```

Comment:
What's the difference between `model.getFilteredModuleList()` and `model.getGradPad().getModuleList()` ?
Is this because there is no need for filtered module List anymore so edit command will just work on the entire module list?

Reply:
yes! idk eh but to me it doesnt make sense if we allow edit via module code, but then we can only edit the modules in the filtered list. I was thinking users can just edit any module from anywhere, what dyou think?

Reply:
Yes I agree with you. I guess because the previous implementation uses the index which is affected by whether it is a filtered list, so now that we are not using index anymore, there should not be a need for filteredModuleList. 👍